traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/edge/Node.java

```diff
@@ -200,9 +200,18 @@ public InetAddress getIp6() {
 
 	@SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.NPathComplexity"})
 	public void setState(final JsonNode state) {
-		final boolean isAvailable = JsonUtils.optBoolean(state, "isAvailable", true);
-		final boolean ipv4Available = JsonUtils.optBoolean(state, "ipv4Available", true);
-		final boolean ipv6Available = JsonUtils.optBoolean(state, "ipv6Available", true);
+		final boolean ipv4Available;
+		final boolean ipv6Available;
+		if (state == null) {
+			LOGGER.warn("got null health state for " + fqdn + ". Setting it to unavailable!");
+			isAvailable = false;
+			ipv4Available = false;
+			ipv6Available = false;
+		} else {
+			isAvailable = JsonUtils.optBoolean(state, "isAvailable", true);
+			ipv4Available = JsonUtils.optBoolean(state, "ipv4Available", true);
+			ipv6Available = JsonUtils.optBoolean(state, "ipv6Available", true);
+		}
```

Comment:
Previously, if `state` was `null`, `JsonUtils.optBoolean()` would return the provided default value. So does this produce a different result than

```java
isAvailable = JsonUtils.optBoolean(state, "isAvailable", false);
ipv4Available = JsonUtils.optBoolean(state, "ipv4Available", false);
ipv6Available = JsonUtils.optBoolean(state, "ipv6Available", false);
```
?

Reply:
With the `if` statement we still get truthy defaults in case the TM is <= 4.0.0.

Reply:
The behavior is slightly different if we were to do that instead, because there is a difference between the entire state being null and the state missing one of the `ipv4Available` or `ipv6Available` keys (which were added in 4.0 I believe). Upon upgrading, those two would need to default to `true` if missing (since they're new); otherwise, TR would mistakenly treat every cache as unavailable.

Reply:
Yep, that is correct.